pkg/controller/watermarkpodautoscaler/replica_calculator.go

```diff
@@ -177,25 +177,26 @@ func (c *ReplicaCalculator) GetResourceReplicas(logger logr.Logger, target *auto
 
 func getReplicaCount(logger logr.Logger, currentReplicas, currentReadyReplicas int32, wpa *v1alpha1.WatermarkPodAutoscaler, name string, adjustedUsage float64, lowMark, highMark *resource.Quantity) (replicaCount int32, utilization int64) {
 	utilizationQuantity := resource.NewMilliQuantity(int64(adjustedUsage), resource.DecimalSI)
-	adjustedHM := float64(highMark.MilliValue()) + wpa.Spec.Tolerance*float64(highMark.MilliValue())
-	adjustedLM := float64(lowMark.MilliValue()) - wpa.Spec.Tolerance*float64(lowMark.MilliValue())
+	adjustedHM := float64(highMark.MilliValue() + highMark.MilliValue()*wpa.Spec.Tolerance.MilliValue()/1000)
```

Comment:
should we check first that `wpa.Spec.Tolerance` is not `nil` ?

Reply:
we will be okay with the defaulting and as it's not a pointer `Tolerance resource.Quantity`
and in `IsDefaultWatermarkPodAutoscaler` we check if the value is == 0, in which cas we default it to 10%.
I added a test case to be explicit about the fact that not specifying the tolerance does not make the spec invalid.
It is however the role of the defaulting methods to ensure we set it to `defaultTolerance`